### DISCUSSION TITLE
Install only CLI features (or have a Python wrapper)?

### DISCUSSION THREAD

**Discussion by averdones** [CBICA/CaPTk#1491] [Q&A]
Hi.

Thank you for the work on this tool.

Is it possible to install only the CLI features of the tool?

I wonder if having a lightweight Python wrapper of some of the CLI commands would be feasible.

**Comment by sarthakpati** [comment]
Thank you for your interest in CaPTk!

Regarding CLI, I assume you have seen this? https://cbica.github.io/CaPTk/Getting_Started.html#gs_commandLineUsage

I'll leave detailed clarification for @AlexanderGetka-cbica

**Comment by AlexanderGetka-cbica** [comment]
Hi @averdones, thanks for your interest!
We don't currently have an option to install only CLI features without the GUI. 

Most of the bundle is taken up by libraries/binaries shared among the CLI and GUI, however, so I am not certain how much space would be saved by removing the GUI elements.

Regarding a Python wrapper, there are a couple ways of doing this that vary in difficulty. Creating a full python package distribution that takes care of installation may be quite difficult at this point. But writing a wrapper that works on an existing installation of CaPTk should be fairly easy to do.

Out of curiosity, could you share with us more about your use case? I can discuss with the team and we can think about this as a potential direction.

**Comment by averdones** [comment]
Thank you for the quick replies @AlexanderGetka-cbica @sarthakpati!

Indeed, my idea was to write a simple wrapper, probably with `subprocess`, around the CLI commands of CaPTk. I can imagine that building a full python package would be an enormous effort.

> Out of curiosity, could you share with us more about your use case? I can discuss with the team and we can think about this as a potential direction.

Sure. I am currently working on a Brain Tumor segmentation project. I am aware that CaPTk can run a pre-trained segmentation model, but I would like to be able to run other models too.

Thus, I am particularly interested in the BraTS Pre-processing Pipeline, particularly the _co-registration_ and the _skull stripping_ steps. Ideally, I would like to import some functions in python after doing a simple pip install. Very roughly, something like this:

```
pip install captk
```
```python
from captk import coregistration, skull_strip

coregistration("t1.nii", "t2.nii", "t1ce.nii", "flair.nii")
skull_strip("t1_coreg.nii", "t2_coreg.nii", "t1ce_coreg.nii", "flair_coreg.nii")
```


People at my department have previously used other tools such as [FSL](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FslInstallation) through a python wrapper called [Nipype](https://nipype.readthedocs.io/en/latest/) (which is analogous to what I am thinking of doing using CaPTk) for coregistration and [ANTs](https://github.com/ANTsX/ANTs) for skull-stripping. I am not sure how these tools compare with CaPTk in terms of performance, but they all require not so simple steps to install and run: some require running bash files and installing external dependencies.

While I haven't found any tool that satisfies all my requirements, I found CaPTk to be the one closest to it. I can imagine copying the self-contained folder of CaPTk into a docker container and using it from there with some simple python CLI wrapper. However, I thought of asking in case this use case had come up before.
